packages/hoprd/crates/hoprd-keypair/src/key_pair.rs

```diff
@@ -0,0 +1,476 @@
+use crate::{
+    errors::{KeyPairError, Result},
+    keystore::{CipherparamsJson, CryptoJson, EthKeystore, KdfType, KdfparamsType},
+};
+use aes::{
+    cipher::{self, InnerIvInit, KeyInit, StreamCipherCore},
+    Aes128,
+};
+use core_crypto::types::{OffchainPublicKey, PublicKey};
+use getrandom::getrandom;
+use hex;
+use scrypt::{scrypt, Params as ScryptParams};
+use serde_json::{from_str as from_json_string, to_string as to_json_string};
+use sha3::{digest::Update, Digest, Keccak256};
+use std::fmt::Debug;
+use utils_log::error;
+use utils_types::traits::ToHex;
+use uuid::Uuid;
+
+#[cfg(all(feature = "wasm", not(test)))]
+use real_base::file::wasm::{metadata, read_to_string, write};
+
+#[cfg(any(not(feature = "wasm"), test))]
+use real_base::file::native::{metadata, read_to_string, write};
+
+const HOPR_CIPHER: &str = "aes-128-ctr";
+const HOPR_KEY_SIZE: usize = 32usize;
+const HOPR_IV_SIZE: usize = 16usize;
+const HOPR_KDF_PARAMS_DKLEN: u8 = 32u8;
+const HOPR_KDF_PARAMS_LOG_N: u8 = 13u8;
+const HOPR_KDF_PARAMS_R: u32 = 8u32;
+const HOPR_KDF_PARAMS_P: u32 = 1u32;
+
+const PACKET_KEY_LENGTH: usize = 32;
+const CHAIN_KEY_LENGTH: usize = 32;
+
+pub type PacketKey = [u8; PACKET_KEY_LENGTH];
+pub type ChainKey = [u8; CHAIN_KEY_LENGTH];
+
+struct Aes128Ctr {
+    inner: ctr::CtrCore<Aes128, ctr::flavors::Ctr128BE>,
+}
+
+impl Aes128Ctr {
+    fn new(key: &[u8], iv: &[u8]) -> std::result::Result<Self, cipher::InvalidLength> {
+        let cipher = aes::Aes128::new_from_slice(key).unwrap();
+        let inner = ctr::CtrCore::inner_iv_slice_init(cipher, iv).unwrap();
+        Ok(Self { inner })
+    }
+
+    fn apply_keystream(self, buf: &mut [u8]) {
+        self.inner.apply_keystream_partial(buf.into());
+    }
+}
+
+#[cfg_attr(feature = "wasm", wasm_bindgen::prelude::wasm_bindgen(getter_with_clone))]
+pub struct IdentityOptions {
+    pub initialize: bool,
+    pub id_path: String,
+    pub password: String,
+    pub use_weak_crypto: Option<bool>,
+    pub private_key: Option<Box<[u8]>>,
+}
+
+#[cfg_attr(feature = "wasm", wasm_bindgen::prelude::wasm_bindgen)]
+impl IdentityOptions {
+    #[cfg_attr(feature = "wasm", wasm_bindgen::prelude::wasm_bindgen(constructor))]
+    pub fn new(
+        initialize: bool,
+        id_path: String,
+        password: String,
+        use_weak_crypto: Option<bool>,
+        private_key: Option<Box<[u8]>>,
+    ) -> Self {
+        Self {
+            initialize,
+            id_path,
+            password,
+            use_weak_crypto,
+            private_key,
+        }
+    }
+}
+
+pub struct HoprKeys {
+    packet_key: (PacketKey, OffchainPublicKey),
+    chain_key: (ChainKey, PublicKey),
+}
```

Comment:
Will replace these with `ChainKeypair` and `OffchainKeypair` that were introduced in my PR.

Reply:
Nice! But please don't make it a Box

Reply:
That was done previously due to wasm_bindgen. Since much of the code is migrated at the moment,  I'm changing all fixed size types to arrays.